subscriptions/providers/apple_in_app/__init__.py

```diff
@@ -127,12 +129,73 @@ def _get_plan_for_product_id(self, product_id: str) -> Plan:
         }
         return Plan.objects.get(**search_kwargs)
 
-    def _get_latest_transaction(self, original_transaction_id: str) -> SubscriptionPayment:
+    def _get_latest_transaction(self, original_transaction_id: str) -> Optional[SubscriptionPayment]:
+        # Latest transaction could be "CANCELLED", as it might be an upgrade.
+        # We assume that the user has a single subscription active for this app on the Apple platform.
         return SubscriptionPayment.objects.filter(
             provider_codename=self.codename,
             metadata__original_transaction_id=original_transaction_id,
         ).order_by('subscription_end').last()
 
+    def _get_active_transaction(self, transaction_id: str, original_transaction_id: str) -> SubscriptionPayment:
+        """
+        Note: this should be a simple `get`, however the database could contain duplicates.
+        Once all duplicates are defeated, replace this with `get`.
```

Comment:
Maybe auto-defeat duplicates using data migration? And write this in a non-tricky way.

Reply:
That's a great idea! I'll check how this can be done.

Reply:
I've added a migration that tries to remove duplicates in as-simple-as-possible manner. Still kinda tricky and there are a few cases in which it just asserts saying "do it yourself, mr smarty pants". I've also included tests for this migration, so there's a remote chance that it will actually work in these non-tricky cases.